src/writeData/components/AddPluginToConfiguration.tsx

```diff
@@ -30,84 +41,122 @@ interface AddPluginToConfigurationCTAProps {
-                      history.push(
-                        `/${ORGS}/${orgID}/load-data/${TELEGRAF_PLUGINS}/${contentID}/new`
-                      )
-                    }}
-                    selected={false}
-                    testID="create-new-configuration-from-plugin--dropdown-item"
-                  >
-                    <span>Create a new configuration</span>
-                  </Dropdown.Item>
-                  <Dropdown.Item
-                    key="Add-to-existing-configuration-telegraf-plugin"
-                    value="Add to an existing configuration"
-                    onClick={() => {}}
-                    selected={false}
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & AddPluginToConfigurationCTAProps
+
+const AddPluginToConfigurationCTAComponent: FC<Props> = props => {
+  const {
+    contentID,
+    fetchTelegrafs,
+    history,
+    orgID,
+    pageContent,
+    telegrafs,
+    thumbnail,
+  } = props
+
+  useEffect(() => {
+    fetchTelegrafs()
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  return (
+    <>
+      <div className="write-data--details">
+        <Grid>
+          <Grid.Row>
+            <Grid.Column
+              widthXS={Columns.Twelve}
+              widthMD={Columns.Six}
+              widthLG={Columns.Three}
+            >
+              <div className="write-data--details-thumbnail">{thumbnail}</div>
+              <Dropdown
+                button={(active, onClick) => (
+                  <Dropdown.Button
+                    active={active}
+                    color={ComponentColor.Primary}
+                    style={{width: '100%'}}
+                    onClick={onClick}
                   >
-                    <span>Add to an existing configuration</span>
-                  </Dropdown.Item>
-                </Dropdown.Menu>
-              )}
-              style={{width: 'auto'}}
-              testID="add-plugin-to-configuration--dropdown"
-            />
-          </Grid.Column>
-          <Grid.Column
-            widthXS={Columns.Twelve}
-            widthMD={Columns.Six}
-            widthLG={Columns.Nine}
-          >
-            <div
-              className="write-data--details-content markdown-format"
-              data-testid="load-data-details-content"
+                    Use this plugin
+                  </Dropdown.Button>
+                )}
+                className="use-plugin--telegraf-configuration"
+                menu={onCollapse => (
+                  <Dropdown.Menu onCollapse={onCollapse}>
+                    <Dropdown.Item
+                      key="Create-new-configuration-telegraf-plugin"
+                      value="Create a new configuration"
+                      onClick={() => {
+                        history.push(
+                          `/${ORGS}/${orgID}/load-data/${TELEGRAF_PLUGINS}/${contentID}/new`
+                        )
+                      }}
```

Comment:
this is another place that will cause re-renders since this anonymous function gets re-created every render loop. I'm not sure if there's a way around it since it depends on `contentID`, but I just wanted to explain this thing.

It _might_ re-render less if the anonymous function is pulled out into a named function like:

```ts
const createNewPlugin = function createNewPlugin() {
  history.push(`/${ORGS}/${orgID}/load-data/${TELEGRAF_PLUGINS}/${contentID}/new`)
}
```
and then passing `createNewPlugin` to `onClick`. probably not worth implementing here, but something I'd like all of us to start thinking about when writing code.

Reply:
Updated.